src/helpers/dimensions.ts

```diff
@@ -255,6 +264,48 @@ export function getHeadlinesDimensions(): AFM.IDimension[] {
     return [{ itemIdentifiers: ["measureGroup"] }];
 }
 
+export function getGeoChartDimensions(
+    mdObject: VisualizationObject.IVisualizationObjectContent,
+): AFM.IDimension[] {
+    const location: VisualizationObject.IBucket = mdObject.buckets.find(
+        bucket => bucket.localIdentifier === LOCATION,
+    );
+    const segment: VisualizationObject.IBucket = mdObject.buckets.find(
+        bucket => bucket.localIdentifier === SEGMENT,
+    );
+    const hasNoSegments = !segment || !segment.items || segment.items.length === 0;
+
+    if (hasNoSegments) {
+        return [
+            {
+                itemIdentifiers: [MEASUREGROUP],
+            },
+            {
+                itemIdentifiers: ((location && location.items) || []).map(
+                    (l: VisualizationObject.IVisualizationAttribute) =>
+                        l.visualizationAttribute.localIdentifier,
+                ),
+            },
+        ];
+    }
+
+    return [
+        {
+            itemIdentifiers: ((segment && segment.items) || []).map(
+                (s: VisualizationObject.IVisualizationAttribute) => s.visualizationAttribute.localIdentifier,
+            ),
+        },
+        {
+            itemIdentifiers: ((location && location.items) || [])
+                .map(
+                    (t: VisualizationObject.IVisualizationAttribute) =>
+                        t.visualizationAttribute.localIdentifier,
+                )
+                .concat([MEASUREGROUP]),
+        },
+    ];
```

Comment:
You might refer to this version, https://github.com/gooddata/gooddata-react-components/pull/1312/files#diff-e34f10013302f162e9f06f5262f59439R306